pkg/actuators/machine/loadbalancers.go

```diff
@@ -154,3 +167,20 @@ func gatherLoadBalancerTargetGroups(client awsclient.Client, names []string) ([]
 
 	return targetGroups, nil
 }
+
+func gatherLoadBalancerTargetGroupRegisteredTargets(client awsclient.Client, targetGroupArn *string) (map[string]struct{}, error) {
```

Comment:
i found the details of this function a little confusing. mainly the disconnect between looking for registered targets and the target health requests inside the function. perhaps a comment about the linkage would be helpful, eg "we use target health to find the registered targets because x,y,z".

Reply:
Fair point, have added a comment at the top of the function to explain why we are using this, lmk if the new comment makes sense

Reply:
that is awesome, thank you!
/lgtm